moto/awslambda/policy.py

```diff
@@ -8,15 +8,11 @@
 )
 from moto.moto_api._internal import mock_random
 
-if TYPE_CHECKING:
-    from .models import LambdaFunction
-
-
 TYPE_IDENTITY = TypeVar("TYPE_IDENTITY")
 
 
 class Policy:
-    def __init__(self, parent: "LambdaFunction"):
+    def __init__(self, parent):  # type: ignore[no-untyped-def]
```

Comment:
Can we improve the typing here and make it a `Union["LambaFunction", "LayerVersion"]`? If we make it explicit what type we expect, that should remove quite a bit of `type: ignore` statements